### ISSUE TITLE
Training an RNN to generate a sine wave

### ISSUE THREAD

**Issue by mahmutkocak** [tflearn/tflearn#1054]
Checking for the [question](https://github.com/tflearn/tflearn/issues/121) of user @DarylWM 
I tried an LSTM model developed in Keras and it worked. 
In answers, they decided to increase the dataset as a solution.
In my model, it worked without increasing the dataset.
What is the difference between two conditions?

My code is here 
`# Simple example using a recurrent neural network to predict time series values
import keras
from keras.models import Sequential
from keras.layers import LSTM, Dense, Activation, Dropout

from __future__ import division, print_function, absolute_import
import numpy as np
import math
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt

step_radians = 0.01
steps_of_history = 100
steps_in_future = 2
index = 0

x = np.arange(0, 4*math.pi, step_radians)
y = np.sin(x)

# Put the data into the right shape
while (index+steps_of_history+steps_in_future < len(y)):
    window = y[index:index+steps_of_history]
    target = y[index+steps_of_history+steps_in_future]
    if index == 0:
        trainX = window
        trainY = target
    else:
        trainX = np.vstack([trainX, window])
        trainY = np.append(trainY, target)
    index = index+1
trainX.shape = (index, steps_of_history, 1)
trainY.shape = (index, 1)

# Network building
model = Sequential()
layers = {'input': 1, 'hidden1': 128, 'hidden2': 512, 'hidden3': 128, 'output': 1}
model.add(LSTM(return_sequences=True, input_shape=(100, 1), units=64))
model.add(Dropout(0.2))
model.add(LSTM(layers['hidden2'], return_sequences=True))
model.add(Dropout(0.3))
model.add(LSTM(layers['hidden3'], return_sequences=False))
model.add(Dropout(0.2))
model.add(Dense(units=layers['output']))
model.add(Activation("linear"))

# Compiling
model.compile(loss="mse", optimizer="sgd")
model.summary()

# Training
model.fit(trainX, trainY, epochs=15, validation_split=0.1, batch_size=128)

# Prepare the testing data set
# testX = window to use for prediction
# testY = actual value
# predictY = predicted value
index = 0
while (index+steps_of_history+steps_in_future < len(y)):
    window = y[index:index+steps_of_history]
    target = y[index+steps_of_history+steps_in_future]
    if index == 0:
        testX = window
        testY = target
    else:
        testX = np.vstack([testX, window])
        testY = np.append(testY, target)
    index = index+1
testX.shape = (index, steps_of_history, 1)
testY.shape = (index, 1)

# Predict the future values
predictY = model.predict(testX)

# Plot the results
plt.figure(figsize=(20,4))
plt.suptitle('Prediction')
plt.title('History='+str(steps_of_history)+', Future='+str(steps_in_future))
plt.plot(y, 'r-', label='Actual')
plt.plot(predictY, 'gx', label='Predicted')
plt.legend()
`
